### ISSUE TITLE
Warehouse resource always requires changes

### ISSUE THREAD

**Issue by doino-gretchenliev** [Snowflake-Labs/terraform-provider-snowflake#1948]
**Provider Version**

0.68.1

**Terraform Version**

1.5.0

**Describe the bug**

Even after successful apply the warehouse size changes always rapper in the plan:
``` 
# module.123456.snowflake_warehouse.test["4X-Large"] will be updated in-place
  ~ resource "snowflake_warehouse" "test" {
        id                                  = "PRD_4X_LARGE"
        name                                = "PRD_4X_LARGE"
      ~ warehouse_size                      = "4XLARGE" -> "4X-Large"
        # (12 unchanged attributes hidden)
    }
```
It happens only with 2X-Large and larger warehouses.

**Code samples and commands**

Warehouse size string: `4X-Large`

**Comment by bfritz12** [comment]
Confirming that I am seeing the same (although `~ warehouse_size                      = "4XLARGE" -> "X4LARGE"` for my implementation)

**Comment by sfc-gh-swinkler** [comment]
Hmm, curious. @sfc-gh-pbosak worked on a patch recently to address this specific issue. I am wondering why we are still seeing it, even in the latest version. Further investigation will be required. In the meantime, would it be possible for you to remove the warehouse and reimport it?

```
terraform state list 
# list of all your terraform resources
terraform state rm <resource_address>
terraform import <resource_address> wh_name
```

**Comment by emancu** [comment]
I am suffering again from the same problem. What is the point of having multiple names for the same warehouse size? 

🤔 Wouldn't it be better to pick one name only for `2XLARGE`, for example? Either `XXLARGE` or `2XLARGE` but not both.

**Comment by sfc-gh-pbosak** [comment]
The fix was introduced in 0.68.2, please try that version instead. If the error still occurs, then we will try to investigate it further.

**Comment by doino-gretchenliev** [comment]
Thank you. I can confirm the issue is resolved with `0.68.2`.
